experimental/SymmetricIntersections/HomogeneousPolynomialsActions.jl

```diff
@@ -0,0 +1,289 @@
+  for m in matrix_representation(repd)
+    poly_m = [R1toR(R1(reverse(vec(collect(m[i,:]))))) for i in 1:nrows(m)]
+    fd = evaluate(f, poly_m)
+    ok, k = divides(fd, f)
+    if !ok || !is_constant(k)
+      return false
+    end
+  end
+  return true
+end
+
+@doc Markdown.doc"""
+    action_on_polynomial(rep::LinRep, f::S) where S <: MPolyElem_dec -> LinRep
+
+Given a linear representation `rep` on a finite vector space `V` and a semi-invariant
+homogeneous polynomial `f` of degree `d` in the ($\mathbb Z$-graded) polynomial
+algebra `R` associated to `V`, return the 1-dimensional linear representation
+corresponding to the induced action on `f`.
+""" 
+function action_on_polynomial(rep::LinRep, f::S) where S <: MPolyElem_dec
+  @req is_homogeneous(f) "f must be a homogeneous polynomial"
+  R = parent(f)
+  @req base_ring(R) === base_field(representation_ring(rep)) "The coefficient of f are in the wrong field"
+  @req ngens(R) == dimension_representation(rep) "There is no induced action on the polynomial ring of f"
+  R1, R1toR = homogeneous_component(R, 1)
+  F = base_ring(R)
+  repd = dual_representation(rep)
+  coll = eltype(matrix_representation(rep))[]
+  for m in matrix_representation(rep)
+    poly_m = [R1toR(R1(reverse((vec(collect(m[i,:])))))) for i in 1:nrows(m)]
+    fd = evaluate(f, poly_m)
+    ok, k = divides(fd, f)
+    @req ok && is_constant(k) "Polynomial is not semi-invariant"
+    push!(coll, matrix(base_ring(R),1,1,[collect(coefficients(k))[1]])::eltype(matrix_representation(rep)))
+  end
+  return Oscar.SymInt._linear_representation(representation_ring(rep), coll)
+end
+
+@doc Markdown.doc"""
+    is_invariant_ideal(rep::LinRep, I::S)
+                                where S <: MPolyIdeal{<: MPolyElem_dec} -> Bool
+
+Given a linear representation `rep` on a finite vector space `V` and a
+homogeneous polynomial `I` in the ($\mathbb Z$-graded) polynomial algebra `R`
+associated to `V`, return whether `I` is invariant under the induce action on
+`R`.
+
+A homogeneous ideal `I` is invariant under a linear action of a group on its
+associated polynomial algebra `R` if and only if the `d`-homogeneous component
+`I_d` of `I` is invariant under the corresponding action on `R_d`.
+""" 
+function is_invariant_ideal(rep::LinRep, I::S) where S <: MPolyIdeal{<: MPolyElem_dec}
+  R = base_ring(I)
+  @req base_ring(R) === base_field(representation_ring(rep)) "The coefficient of f are in the wrong field"
+  @req ngens(R) == dimension_representation(rep) "There is no induced action on the polynomial ring of I"
+  gene = minimal_generating_set(I)
+  R1, R1toR = homogeneous_component(R, 1)
+  repd = dual_representation(rep)
+  for f in gene for m in matrix_representation(repd)
+    poly_m = [R1toR(R1(reverse(vec(collect(m[i,:]))))) for i in 1:nrows(m)]
+    fd = evaluate(f, poly_m)
+    fd in I || return false
+  end
+  end
+  return true
+end
+
+@doc Markdown.doc"""
+    action_on_ideal(rep::LinRep, I::S)
```

Comment:
```suggestion
    action_on_ideal(rep::LinRep, I::S)
```
similar as above.  `linear_representation` seems like a better name to me.